City-of-Helsinki/berth-reservations-ui
# VEN-1099, VEN-1100 | Tunnistamo login and new navigation

## Description :sparkles:

* Added Tunnistamo authentication service and components using **admin UI environment** for development/testing
* Replaced navigation with HDS Navigation components
* Implemented hot switching language

## Issues :bug:

### Closes :no_good_woman:

* [VEN-1099](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1099): Implement Tunnistamo login for customer UI
* [VEN-1100](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1100): User dropdown and logging out from customer UI

## Testing :alembic:

### Automated tests :gear:️

* Updated browser tests
* Added basic tests for Navbar
* Added tests for authService

### Manual testing :construction_worker_man:

* The navigation should not do anything funky
* Logging in and out (using GitHub, for example) should work
* Test a private route `/fi/auth-test`
  * When logged out, it should redirect the you to login
  * Once logged in, you should be returned to `/fi/auth-test`
* Test a normal route (like `/fi/winter-storage`)
  * Be logged out and click login from the navigation
  * You should be redirected back to the page once logged in
* Switching languages should work instantly and retain the page you are on

## Screenshots :camera_flash:

![image](https://user-images.githubusercontent.com/15479131/110433084-b1304300-80b8-11eb-9d0c-2cb41273f9ab.png)
